kafka-ui-api/src/main/java/com/provectus/kafka/ui/serde/schemaregistry/SchemaRegistryAwareRecordSerDe.java

```diff
@@ -92,14 +93,23 @@ public SchemaRegistryAwareRecordSerDe(KafkaCluster cluster) {
 
   public DeserializedKeyValue deserialize(ConsumerRecord<Bytes, Bytes> msg) {
     try {
-      return new DeserializedKeyValue(
-          msg.key() != null
-              ? getMessageFormatter(msg, true).format(msg.topic(), msg.key().get())
-              : null,
-          msg.value() != null
-              ? getMessageFormatter(msg, false).format(msg.topic(), msg.value().get())
-              : null
-      );
+      var builder = DeserializedKeyValue.builder();
+      if (msg.key() != null) {
+        MessageFormatter messageFormatter = getMessageFormatter(msg, true);
+        builder.key(messageFormatter.format(msg.topic(), msg.key().get()));
+        builder.keyFormat(messageFormatter.getFormat());
+        builder.keySchemaId(getSchemaId(msg.key()).map(String::valueOf).orElse(null));
```

Comment:
if stringFormatter was chosen calling getSchemaId() can cause unexpected schemaId to be returned, maybe we should move getSchemaId() to formatter and return Optional.empty in case of non-schema formatters

Reply:
I see your point but I don't think that the MessageFormatter interface should be aware of schemas. Let me add messageFormatter type check, it will remove problem with stringFormatter.

Reply:
Fixed, please review